integration/sandbox_run_rollback_test.go

```diff
@@ -229,6 +235,97 @@ func TestRunPodSandboxWithShimStartAndTeardownCNIFailure(t *testing.T) {
 	t.Run("JustCleanup", testCase(false))
 }
 
+// TestRunPodSandboxWithShimStartAndTeardownCNISlow should keep the sandbox
+// record if failed to rollback CNI API.
+func TestRunPodSandboxWithShimStartAndTeardownCNISlow(t *testing.T) {
+	if runtime.GOOS != "linux" {
+		t.Skip()
+	}
+	if os.Getenv("ENABLE_CRI_SANDBOXES") != "" {
+		t.Skip()
+	}
+
+	t.Log("Init PodSandboxConfig with specific key")
+	labels := map[string]string{
+		t.Name(): "true",
+	}
+	sbConfig := PodSandboxConfig(t.Name(), "failpoint", WithPodLabels(labels))
+
+	t.Log("Inject CNI failpoint")
+	conf := &failpointConf{
+		// Delay 10s
+		Add: "1*delay(10000)",
+		// This is to make sure the sandbox container won't be deleted even if the defer functions are invoked in sandbox_run.go
+		// Without this, the test is flaky because ListPodSandbox may return empty list.
+		Del: "1*error(please retry)",
+	}
+	injectCNIFailpoint(t, sbConfig, conf)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		t.Log("Create a sandbox")
+		_, err := runtimeService.RunPodSandbox(sbConfig, failpointRuntimeHandler)
+		require.Error(t, err)
+		require.ErrorContains(t, err, "error reading from server: EOF")
+	}()
+
+	time.Sleep(5 * time.Second)
+	t.Log("Restart containerd after 5s")
+	RestartContainerd(t)
```

Comment:
synced with @qiutongs last night. We need to send sigkill to prevent containerd from graceful shutdown. The graceful shutdown will cancel the context to the inflight rpc call, which might invoke the defer to cleanup the resource.